app/services/research_safety_adapters/scishield_training_synchronizer.rb

```diff
@@ -12,20 +12,40 @@ def synchronize
         Rails.logger.error(msg)
         Rollbar.error(msg) if defined?(Rollbar)
       else
+        user_certs = {}
+
+        # In some cases, making many requests in a row results in the API not
+        # responding. Making requests in batches with sleep in between seems
+        # to make the API work in cases where this is a problem. `sleep` is put
+        # first here because `api_unavailable?` will have already made 10 API
+        # requests.
+        users.in_batches(of: batch_size) do |user_batch|
+          sleep(batch_sleep_time)
+
+          user_batch.each do |user|
+            adapter = ScishieldApiAdapter.new(user, api_client)
+            begin
+              cert_names = adapter.certified_course_names_from_api
+              user_certs[user.id.to_s] = cert_names if cert_names.presence
+            rescue => e
+              Rails.logger.error(e.message)
+              Rollbar.error(e.message) if defined?(Rollbar)
+            end
+          end
+        end
+
         ScishieldTraining.transaction do
-          ScishieldTraining.delete_all
+          ScishieldTraining.delete(user_certs.keys.map(&:to_s))
```

Comment:
Following up on https://github.com/Oregon-State/nucore-osu/pull/325#discussion_r1409517777...

I think we should start with the simplest approach - delete everything.  We want the db to reflect the API.  If the records are not there locally we check the API again anyway.  What do you think?